code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -942,7 +942,12 @@ Pass a positive integer as an argument to override a bot's default speed.
 				ejectpai(usr)
 
 /mob/living/simple_animal/bot/update_icon_state()
-	icon_state = "[initial(icon_state)][get_bot_flag(bot_mode_flags, BOT_MODE_ON)]"
+	var/base_state
+	if(!isnull(base_icon_state))
+		base_state = initial(base_icon_state)
+	else
+		base_state = initial(icon_state)
```

Comment:
When will this ever be null?

Reply:
Just an assumption so I don't break existing behavior. For example floor bot does not use  base_icon_state and I'm not sure about other bots